File: _Practice/LeetCode/001. Two Sum.java
// https://leetcode.com/problems/two-sum/
// Time: O(n) in expectation

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> lookUp = new HashMap<>();
        int SIZE = nums.length;
        for (int i = 0; i < SIZE; ++i) {
            if (lookUp.containsKey(target - nums[i])) {
                return new int[]{i, lookUp.get(target - nums[i])};
            } else {
                lookUp.put(nums[i], i);
            }
        }
        return new int[]{-1, -1};
    }
}

File: _Practice/LeetCode/001. Two Sum.cpp
// https://leetcode.com/problems/two-sum/
// Time: O(n) in expectation

class Solution {
public:
    vector<int> twoSum(vector<int> &nums, int target) {
        vector<int> res(2);
        unordered_map<int, int> lookUp;
        int SIZE = nums.size();
        for (int i = 0; i < SIZE; ++i) {
            int complement = target - nums[i];
            if (lookUp.count(complement)) {
                res[0] = i, res[1] = lookUp[complement];
                break;
            } else {
                lookUp.insert({nums[i], i});
            }
        }
        return res;
    }
};
